### PR TITLE
Implement extension support for native file dialog on Android

### PR DESCRIPTION
**Pull request by syntaxerror247**
This PR adds support for filtering files by extension (e.g., `*.jpg, *.txt` ) in the native Android file dialog, bringing Android in line with functionality on other platforms.

Now, Users can  filtering files by extensions (e.g., `*.txt, *.mp4`), file types (e.g., `txt, mp4`), or MIME types (e.g., `text/plain, video/mp4`).

### CODE REVIEW

File: platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt [hunk 1]
```diff
@@ -145,16 +146,54 @@ internal class FilePicker {
...
 			activity?.startActivityForResult(intent, FILE_PICKER_REQUEST)
 		}
+
+		/**
+		 * Retrieves the MIME type for a given file extension.
+		 *
+		 * @param p_input the extension whose MIME type is to be determined.
+		 * @return the MIME type as a string, or "application/octet-stream" if the type is unknown.
+		 */
+		private fun resolveMimeType(p_input: String): String {
```
Comment (m4gr3d) [inline platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt:168]: `p_input` does not match the java/kotlin stype, can you update the parameter name.

File: platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt [hunk 2]
```diff
@@ -145,16 +146,54 @@ internal class FilePicker {
...
+			var input = p_input
+
+			// Fix for extensions like "*.txt", "*.jpg", etc.
+			if (p_input.startsWith("*.")) {
+				input = p_input.substring(2)
+			}
```
Comment (m4gr3d) [inline platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt:174]: Should this also check for `p_input.startsWith(".")`? 
The [documentation](https://developer.android.com/reference/android/webkit/MimeTypeMap#getMimeTypeFromExtension(java.lang.String)) states that the passed parameter should be `A file extension without the leading '.'`

File: platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt [hunk 3]
```diff
@@ -145,16 +146,54 @@ internal class FilePicker {
...
+
+			val mimeTypeMap = MimeTypeMap.getSingleton()
+			val resolvedMimeType = mimeTypeMap.getMimeTypeFromExtension(input)
+			if (resolvedMimeType != null) {
+				return resolvedMimeType
+			}
+			// Check if the input is already a valid MIME type.
+			if (mimeTypeMap.hasMimeType(input)) {
+				return input
+			}
```
Comment (m4gr3d) [inline platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt:184]: Seems like we should do this check at the beginning of the method.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt (hunk 1 above)
Comment (syntaxerror247) [reply platform/android/java/lib/src/org/godotengine/godot/io/FilePicker.kt:168]: I've observed this format being used in other Java/Kotlin files as well, and most importantly, I couldn't come up with a better name 😅.

**Review by m4gr3d — APPROVED** [review]
Looks good!

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**